ratis-server/src/main/java/org/apache/ratis/server/storage/SnapshotManager.java

```diff
@@ -98,17 +102,21 @@ public void installSnapshot(StateMachine stateMachine,
           }
           // create the temp snapshot file and put padding inside
           out = new FileOutputStream(tmpSnapshotFile);
+          digester = MD5Hash.getDigester();
+          outWithDigest = new DigestOutputStream(out, digester);
         } else {
           Preconditions.assertTrue(tmpSnapshotFile.exists());
           out = new FileOutputStream(tmpSnapshotFile, true);
+          outWithDigest = new DigestOutputStream(out, digester);
           FileChannel fc = out.getChannel();
           fc.position(chunk.getOffset());
         }
 
         // write data to the file
-        out.write(chunk.getData().toByteArray());
+        outWithDigest.write(chunk.getData().toByteArray());
```

Comment:
Use
```
        new DigestOutputStream(out, digester.get()).write(chunk.getData().toByteArray());
```
The other changes are not needed.

Reply:
done

Reply:
The lint says new DigestOutputStream(...) here will cause a opened stream, so I' ll use the try() clause